ve/opencl/engine_opencl.cpp

```diff
@@ -92,20 +92,42 @@ EngineOpenCL::EngineOpenCL(const ConfigParser &config, jitk::Statistics &stat) :
                                     work_group_size_3dz(config.defaultGet<int>("work_group_size_3dz", 2)),
                                     compile_flg(config.defaultGet<string>("compiler_flg", "")),
                                     default_device_type(config.defaultGet<string>("device_type", "auto")),
+                                    platform_no(config.defaultGet<int>("platform_no", -1)),
                                     verbose(config.defaultGet<bool>("verbose", false)),
                                     stat(stat) {
     vector<cl::Platform> platforms;
     cl::Platform::get(&platforms);
     if(platforms.size() == 0) {
         throw runtime_error("No OpenCL platforms found");
     }
-    cl::Platform default_platform=platforms[0];
-    if(verbose) {
-        cout << "Using platform: " << default_platform.getInfo<CL_PLATFORM_NAME>() << endl;
+
+    bool found = false;
+    cl::Platform platform;
+    if (platform_no == -1) {
+        for (auto pform : platforms) {
+            // Pick first valid platform
+            try {
+                // Get the device of the platform
+                platform = pform;
+                device = getDevice(platform, default_device_type);
+                found = true;
+            } catch(cl::Error err) {
+                // We try next platform
+            }
+        }
+    } else {
+        platform = platforms[platform_no];
```

Comment:
I think you should add a overflow check of the ``platforms`` vector, i.e. ``platforms.size() > platform_no``